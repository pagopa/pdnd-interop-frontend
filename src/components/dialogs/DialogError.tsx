import useResolveError from '@/hooks/useResolveError'
import { Dialog, DialogContent, DialogTitle, Typography } from '@mui/material'
import React from 'react'
import type { FallbackProps } from 'react-error-boundary'
import { useDialog } from '@/stores'

export const DialogError: React.FC<FallbackProps> = (fallbackProps) => {
  const ariaLabelId = React.useId()
  const ariaDescriptionId = React.useId()

  const { title, description, content } = useResolveError(fallbackProps)
  const { closeDialog } = useDialog()

  return (
    <Dialog
      aria-labelledby={ariaLabelId}
      aria-describedby={ariaDescriptionId}
      open
      onClose={closeDialog}
      fullWidth
    >
      <DialogTitle id={ariaLabelId}>{title}</DialogTitle>
      <DialogContent>
        <Typography id={ariaDescriptionId} sx={{ mb: 2 }}>
          {description}
        </Typography>
        {content}
      </DialogContent>
    </Dialog>
  )
}
