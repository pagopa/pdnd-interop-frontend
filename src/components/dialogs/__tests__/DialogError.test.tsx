import React from 'react'
import { DialogError } from '../DialogError'
import { vi } from 'vitest'
import { render } from '@testing-library/react'

describe('DialogError testing', () => {
  it('should match the snapshot', () => {
    const screen = render(<DialogError error={new Error('test')} resetErrorBoundary={vi.fn()} />)
    expect(screen.baseElement).toMatchSnapshot()
  })
})
