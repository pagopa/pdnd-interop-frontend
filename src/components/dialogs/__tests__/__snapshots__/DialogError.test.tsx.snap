// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogError testing > should match the snapshot 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby=":r1:"
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r0:"
        >
          default.title
        </h2>
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-z2eky3-MuiTypography-root"
            id=":r1:"
          >
            default.description
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            actions.reloadPage
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
